src/precice/bindings/python_future/precice_future.pyx

```diff
@@ -98,134 +98,140 @@ cdef class Interface:
         self.thisptr.setMeshTriangle (mesh_id, first_edge_id, second_edge_id, third_edge_id)
 
     # sets mesh triangle from vertices
-    def set_mesh_triangle_with_edges (self, int mesh_id, int first_vertex_id, int second_vertex_id, int third_vertex_id):
+    def set_mesh_triangle_with_edges (self, mesh_id, first_vertex_id, second_vertex_id, third_vertex_id):
         self.thisptr.setMeshTriangleWithEdges (mesh_id, first_vertex_id, second_vertex_id, third_vertex_id)
 
     # sets mesh quad from edges
-    def set_mesh_quad (self, int mesh_id, int first_edge_id, int second_edge_id, int third_edge_id, int fourth_edge_id):
+    def set_mesh_quad (self, mesh_id, first_edge_id, second_edge_id, third_edge_id, fourth_edge_id):
         self.thisptr.setMeshQuad (mesh_id, first_edge_id, second_edge_id, third_edge_id, fourth_edge_id)
 
     # sets mesh quad from vertices
-    def set_mesh_quad_with_edges (self, int mesh_id, int first_vertex_id, int second_vertex_id, int third_vertex_id, int fourth_vertex_id):
+    def set_mesh_quad_with_edges (self, mesh_id, first_vertex_id, second_vertex_id, third_vertex_id, fourth_vertex_id):
         self.thisptr.setMeshQuadWithEdges (mesh_id, first_vertex_id, second_vertex_id, third_vertex_id, fourth_vertex_id)
 
     # data access
     # hasData
-    def has_data (self, str data_name, int mesh_id):
+    def has_data (self, str data_name, mesh_id):
         return self.thisptr.hasData(convert(data_name), mesh_id)
 
-    def get_data_id (self, str data_name, int mesh_id):
+    def get_data_id (self, str data_name, mesh_id):
         return self.thisptr.getDataID (convert(data_name), mesh_id)
 
-    def map_read_data_to (self, int to_mesh_id):
+    def map_read_data_to (self, to_mesh_id):
         self.thisptr.mapReadDataTo (to_mesh_id)
 
-    def map_write_data_from (self, int from_mesh_id):
+    def map_write_data_from (self, from_mesh_id):
         self.thisptr.mapWriteDataFrom (from_mesh_id)
 
-    def write_block_vector_data (self, int data_id, value_indices, values):
+    def write_block_vector_data (self, data_id, value_indices, values):
+        size, dimensions = values.shape
+        assert(dimensions == self.get_dimensions())
         cdef np.ndarray[int, ndim=1] _value_indices = np.ascontiguousarray(value_indices, dtype=np.int32)
-        cdef np.ndarray[double, ndim=1] _values = np.ascontiguousarray(values, dtype=np.double)
-        assert(_values.size / self.get_dimensions() == _value_indices.size)
+        cdef np.ndarray[double, ndim=1] _values = np.ascontiguousarray(values.flatten(), dtype=np.double)
+        assert(size == _value_indices.size)
         size = value_indices.size
         self.thisptr.writeBlockVectorData (data_id, size, &_value_indices[0], &_values[0])
 
-    def write_vector_data (self, int data_id, value_index, value):
+    def write_vector_data (self, data_id, value_index, value):
+        dimensions = value.size
```

Comment:
Should be able to handle tuples and lists

Reply:
done.